c7n/resources/health.py

```diff
@@ -11,41 +11,121 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from cStringIO import StringIO
-import csv
-import datetime
-from datetime import timedelta
-from dateutil.parser import parse
-from dateutil.tz import tzutc
-import time
-from botocore.exceptions import ClientError
-
-from c7n.actions import BaseAction, ActionRegistry
-from c7n.filters import ValueFilter, Filter, OPERATORS, FilterRegistry
+
+from c7n.actions import (
+    ActionRegistry, BaseAction
+)
+from c7n.filters import (
+    FilterRegistry, AgeFilter, ValueFilter, Filter, OPERATORS
+)
+
+from c7n.query import QueryResourceManager
 from c7n.manager import resources
-from c7n.query import ResourceManager
-from c7n.utils import local_session, type_schema
 
+from c7n import utils
+from c7n.utils import type_schema
 
-def get_health_events(session_factory):
-    session = local_session(session_factory)
-    client = session.client('health')
-    p = client.get_paginator('describe_events')
-    return [e['events'] for e in p.paginate()]
+filters = FilterRegistry('health.actions')
+actions = ActionRegistry('health.filters')
 
 
-@resources.register('health')
-class HealthEvents(ResourceManager):
-    """Resource manager for AWS health events"""
+@resources.register('health-events')
+class HealthEvents(QueryResourceManager):
+    """Query resource manager for AWS health events"""
 
     class resource_type(object):
         service = 'health'
+        type = 'event'
+        enum_spec = ('describe_events', 'events', None)
+        name = 'eventTypeCode'
+        id = 'arn'
+        filter_name = None
+        filter_type = None
+        dimension = None
+        date = 'startTime'
+        config_type = 'AWS::Health::Event'
+
+    filter_registry = filters
+    action_registry = actions
+
+    def __init__(self, ctx, data):
+        super(HealthEvents, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(self.data.get('query', []))
```

Comment:
this needs a sensible default, we can't default to querying with no params, when they keep effectively lifetime historical that i've seen.